new-dti-website/components/team/MemberGroup.tsx

```diff
@@ -0,0 +1,326 @@
+import { Dispatch, SetStateAction, useMemo, useState, RefObject } from 'react';
+import Image from 'next/image';
+import { Card } from '../ui/card';
+import { ibm_plex_mono } from '../../src/app/layout';
+import teamRoles from './data/roles.json';
+import subteams from './data/subteams.json';
+import connectIcons from './data/connectIcons.json';
+import { getFullRoleFromDescription } from '../../src/app/utils';
+import useScreenSize from '../../src/hooks/useScreenSize';
+
+type MemberSummaryProps = {
+  image: string;
+  firstName: string;
+  lastName: string;
+  role: string;
+  roleDescription: RoleDescription;
+  enlarged: boolean;
+};
+
+const MemberSummary: React.FC<MemberSummaryProps> = ({
+  image,
+  firstName,
+  lastName,
+  role,
+  roleDescription,
+  enlarged
+}) => {
+  const chipColor = teamRoles[role as Role].color;
+
+  return (
+    <div id="memberCard" className="flex flex-col md:gap-3 xs:gap-2">
+      <img
+        src={image}
+        alt={`${firstName}-${lastName}`}
+        className={`rounded-md h-auto ${enlarged ? 'w-[244px]' : 'w-[202px]'}`}
+      />
+      <h3
+        className={`xs:text-lg font-${enlarged ? 'semibold md:text-2xl' : 'bold md:text-[22px]'}`}
+      >{`${firstName} ${lastName}`}</h3>
+      <p
+        className={`w-fit px-3 py-1 rounded-2xl ${ibm_plex_mono.className} md:text-sm xs:text-xs`}
+        style={{ backgroundColor: chipColor }}
+      >
+        {getFullRoleFromDescription(roleDescription)}
+      </p>
+    </div>
+  );
+};
+
+type MemberCardProps = {
+  onClick: () => void;
+  firstName: string;
+  lastName: string;
+  role: string;
+  image: string;
+  roleDescription: RoleDescription;
+  cardState: number | undefined;
+};
+
+const MemberCard: React.FC<MemberCardProps> = (props: MemberCardProps) => (
+  <Card
+    id="memberCard"
+    className={`w-fit md:p-3 md:pb-4 xs:p-2 xs:pb-3 h-fit justify-self-center ${
+      props.cardState ? 'opacity-70 hover:opacity-100' : 'opacity-100'
+    } ${
+      props.cardState === 0 && 'shadow-[0_4px_4px_0_#00000040]'
+    } hover:shadow-[0_4px_4px_0_#00000040]`}
+    onClick={props.onClick}
+  >
+    <MemberSummary {...props} enlarged={false} />
+  </Card>
+);
+
+type MemberDetailsProps = {
+  onClose: () => void;
+  firstName: string;
+  lastName: string;
+  role: string;
+  roleDescription: RoleDescription;
+  graduation: string;
+  major: string;
+  hometown: string;
+  about: string;
+  subteams: readonly string[];
+  email: string;
+  website?: string | null;
+  linkedin?: string | null;
+  image: string;
+};
+
+type Subteam =
+  | 'courseplan'
+  | 'reviews'
+  | 'queuemein'
+  | 'cuapts'
+  | 'idol'
+  | 'cornellgo'
+  | 'carriage'
+  | 'leads'
+  | 'business'
+  | 'zing'
+  | 'dac';
+
```

Comment:
I'd remove this type. We can just set it to a string (Subteam is not a fixed type).